libs/sdk-react-native/ios/RNBreezSDK.swift

```diff
@@ -450,10 +450,9 @@ class RNBreezSDK: RCTEventEmitter {
     }
 
     @objc(buyBitcoin:resolver:rejecter:)
-    func buyBitcoin(_ provider: String, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) -> Void {
+    func buyBitcoin(_ req:[String: Any], resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) -> Void {
         do {
-            let buyBitcoinProvider = try BreezSDKMapper.asBitcoinProvider(provider: provider)
-            let result = try getBreezServices().buyBitcoin(provider: buyBitcoinProvider)
+            let result = try getBreezServices().buyBitcoin(req: req)
             resolve(result)        
         } catch let err {
             rejectErr(err: err, reject: reject)
```

Comment:
The req needs to be converted into a BuyBitcoinRequest type
```
        if let buyBitcoinRequest = BreezSDKMapper.asBuyBitcoinRequest(reqData: req) {
            do {
                let response = try getBreezServices().buyBitcoin(req: buyBitcoinRequest)                
                resolve(BreezSDKMapper.dictionaryOf(buyBitcoinResponse: response))
            } catch let err {
                reject(RNBreezSDK.TAG, "Error calling buyBitcoin \(err)", err)
            }
        } else {
            rejectErr(err: SdkError.Generic(message:"Invalid reqData"), reject: reject)            
        }
```

Reply:
Done as part of 4efe618